VideoAPI/Video.API/Controllers/VideoEventsController.cs

```diff
@@ -61,20 +58,57 @@ public async Task<IActionResult> PostEventAsync(ConferenceEventRequest request)
                 return NoContent();
             }
 
-            var callbackEvent = new CallbackEvent
+            var callbackEvent = MapEventRequestToEventHandlerDto(conferenceId, participantId, request);
+            await _eventHandlerFactory.Get(request.EventType).HandleAsync(callbackEvent);
+            return NoContent();
+        }
+
+        private SaveEventCommand MapEventRequestToEventCommand(Guid conferenceId, ConferenceEventRequest request)
+        {
+            GetRoomTypeEnums(request, out var transferTo, out var transferFrom);
+
+            var command = new SaveEventCommand(conferenceId, request.EventId, request.EventType,
+                request.TimeStampUtc, transferFrom, transferTo, request.Reason, request.Phone);
+            if (Guid.TryParse(request.ParticipantId, out var participantId))
+            {
+                command.ParticipantId = participantId;
+            }
+
+            command.TransferredFromRoomLabel = request.TransferFrom;
+            command.TransferredToRoomLabel = request.TransferTo;
+
+            return command;
+        }
+
+        private CallbackEvent MapEventRequestToEventHandlerDto(Guid conferenceId, Guid participantId,
+            ConferenceEventRequest request)
+        {
+            GetRoomTypeEnums(request, out var transferTo, out var transferFrom);
+            
+            return new CallbackEvent
             {
                 EventId = request.EventId,
                 EventType = request.EventType,
                 ConferenceId = conferenceId,
                 Reason = request.Reason,
-                TransferTo = request.TransferTo,
-                TransferFrom = request.TransferFrom,
+                TransferTo = transferTo,
+                TransferFrom = transferFrom,
                 TimeStampUtc = request.TimeStampUtc,
                 ParticipantId = participantId,
-                Phone = request.Phone
+                Phone = request.Phone,
+                TransferredFromRoomLabel = request.TransferFrom,
+                TransferredToRoomLabel = request.TransferTo
             };
-            await _eventHandlerFactory.Get(request.EventType).HandleAsync(callbackEvent);
-            return NoContent();
+        }
+
+        private static void GetRoomTypeEnums(ConferenceEventRequest request, out RoomType? transferTo,
```

Comment:
why static?

Reply:
it's reshaper/rider's default implementation when you introduce a method.